src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package task;
+import java.util.Date;
+import java.text.ParseException;
+
+public class Deadline extends Task{
+    private Date deadline;
+    public Deadline(String description, String deadline) throws ParseException {
+        super(description);
+        this.deadline = super.readFormat.parse(deadline);
+    }
+    @Override
+    public String save() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[D]");
+        sb.append(super.toString()).append(" (by: ");
+        sb.append(super.readFormat.format(this.deadline));
+        sb.append(")");
```

Comment:
This also seems similar to line 22-27 as well. Perhaps creating a helper method will help

Reply:
This was to ensure that the task was saved according to the read format and not the write format, hence the similar but importantly different method.